src/engine/mod.rs

```diff
@@ -0,0 +1,189 @@
+// This file is part of rust-web/twig
+//
+// For the copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+//! The Twig Engine.
+
+mod template_cache;
+use loader::Loader;
+use std::rc::Rc;
+use template;
+use error::ErrorCode;
+
+pub mod error;
+pub mod options;
+pub mod setup;
+pub mod extension_registry;
+pub mod parser;
+pub mod node;
+pub use self::node::Node;
+pub use self::parser::{Parser, lexer, Lexer};
+pub use self::error::{TwigError, TwigErrorCode, ExtensionRegistryError, ExtensionRegistryErrorCode};
+pub use self::options::Options;
+pub use self::setup::Setup;
+pub use self::extension_registry::ExtensionRegistry;
+
+
+#[derive(Default, Debug)]
+pub struct Engine {
+    options: Options,
+    ext: Option<Rc<ExtensionRegistry>>,
```

Comment:
I think it is possible to get rid of it by having Extension -> Registry -> Options pipeline.